Monkey patch activerecord module to check pending migration

peiyee opened the pull request:
<!--
Thanks for creating a Pull Request! Before you submit, please make sure you've done the following:

- I read the contributing document at https://rubyonjets.com/docs/contributing/
-->

<!--
Make our lives easier! Choose one of the following by uncommenting it:
-->

This is a 🐞 bug fix.
<!-- This is a 🙋‍♂️ feature or enhancement. -->
<!-- This is a 🧐 documentation change. -->

<!--
Before you submit this pull request, make sure to have a look at the following checklist. To mark a checkbox done, replace [ ] with [x]. Or after you create the issue you can click the checkbox.

If you don't know how to do some of these, that's fine!  Submit your pull request and we will help you out on the way.
-->

- [ ] I've added tests (if it's a bug, feature or enhancement)
- [ ] I've adjusted the documentation (if it's a feature or enhancement)
- [x] The test suite passes (run `bundle exec rspec` to verify this)

## Summary

Created a new module `lib/jets/commands/templates/skeleton/lib/core_extensions/active_record/migration_checker.rb`
which contains method `prepare_test_db` 
This method will be used in spec_helper to check any pending migration.

Using this new method instead of reusing `ActiveRecord::Migration.load_schema_if_pending!` is due to the method has hardcoded db command `bin/rails db:test:prepare` and error message

## Context

https://github.com/tongueroo/jets/issues/337

## Version Changes

<!--
Which semantic version change would you recommend?
If you don't know, feel free to omit it.
-->

tongueroo commented on lib/jets/commands/templates/skeleton/lib/core_extensions/active_record/migration_checker.rb:
```diff
@@ -0,0 +1,37 @@
+  module ActiveRecord
+    module MigrationChecker
+      def prepare_test_db
+        require 'rails'
```
Wondering how to do this w/o requiring rails which will define the Rails const 🤔. Related: https://community.rubyonjets.com/t/why-does-jets-define-rails-module/29

Also, wondering how to do it w/o generating a file as part of the skeleton template 🧐

peiyee replied:
@tongueroo thanks for your review

> Wondering how to do this w/o requiring rails which will define the Rails const

Should be fixed in this ffe6e61f1584033429da5b87b551998c2233d608

> how to do it w/o generating a file as part of the skeleton template

where would you recommend to put the migration checker? I have tried to put it inside the `lib/jets/spec_helpers.rb` (c09c2abd3365d9e3f2cc643c278c208eb800ecdb) but it fails the [build](https://circleci.com/gh/tongueroo/jets/2678)

tongueroo replied:
> RE: where would you recommend to put the migration checker?

Unsure. Spec helpers seems like a fine place. Maybe hack it and don’t run it as part of the spec run?

> RE: Not generating the file as part of the skeleton

Thanks for changing that. It’s better to have more control of the overrides in case there are changes in the future. 

Would also move it from lib/core_ext to overrides/rails. Have the overrides/monkey patching close to each other.

peiyee replied:
I have move it into `lib/jets/overrides/active_record` because most of the code is coming from active record module. Also i have added a new ENV config `SKIP_MIGRATION_CHECK` to control whether we should run the migration checking or not.
Please help to take a look again. :smile:

peiyee commented:
Hi @tongueroo is this good to merge?

tongueroo commented on lib/jets/core_ext/overrides/active_record/migration_checker.rb:
```diff
@@ -0,0 +1,32 @@
+module Overrides
+  module ActiveRecord
+    module MigrationChecker
```
Not exactly sure what happened. Looks like this is still in jets/core_ext instead of jets/overrides 🤔

tongueroo replied:
Think the review comment wasn’t surfacing. Anyway, just resubmitted it and double checked.

peiyee replied:
noted, i will look into this tonight.

peiyee commented:
> Not exactly sure what happened. Looks like this is still in jets/core_ext instead of jets/overrides

Yeah, it was my mistake previously. I have updated the code, and rebased with the latest master.

tongueroo commented:
Here are some QA notes for posterity. Generated a new app with the `peiyee:fix-app-rspec-migration-warning` branch. Then ran these commands to test.

```
$ jets generate scaffold post title:string
      invoke  active_record
      create    db/migrate/20190925170012_create_posts.rb
      create    app/models/post.rb
      invoke  resource_route
       route    resources :posts
      invoke  scaffold_controller
      create    app/controllers/posts_controller.rb
      invoke    erb
      create      app/views/posts
      create      app/views/posts/index.html.erb
      create      app/views/posts/edit.html.erb
      create      app/views/posts/show.html.erb
      create      app/views/posts/new.html.erb
      create      app/views/posts/_form.html.erb
      invoke    helper
      create      app/helpers/posts_helper.rb
$ bundle exec rspec
Migrations are pending. To resolve this issue, run:

        jets db:migrate
$ jets db:migrate
== 20190925170012 CreatePosts: migrating ======================================
-- create_table(:posts)
   -> 0.0048s
== 20190925170012 CreatePosts: migrated (0.0049s) =============================

$ bundle exec rspec
No examples found.

Finished in 0.00044 seconds (files took 3.96 seconds to load)
0 examples, 0 failures

$ ```

tongueroo commented:
Released in [v2.2.4](https://github.com/tongueroo/jets/blob/master/CHANGELOG.md)
